### PR TITLE
Move to github action

### PR DESCRIPTION
**Pull request by alain-andre**
Move image creation from Travis to Github Action.

### CODE REVIEW

**Comment by senhalil** [comment]
- [ ] Needs github secrets to be defined

**Comment by alain-andre** [comment]
> [x] Needs github secrets to be defined

They are now defined at group level. But each fork must do it by its own.

File: .github/actions/build.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [[ $GITHUB_REF == *tags* ]]; then
+  TAG=${GITHUB_REF#refs/tags/}
+else
+  echo "> Building image for dev purpose (${REPOSITORY}) with or-tools ${ORTOOLS_VERSION}"
+  TAG=$(echo "${REPOSITORY}" | sed "s/\/.*//")
```
Comment (braktar) [inline .github/actions/build.sh:9]: The current TAG is only based on the repository name when there is no tag.
This means that if we have multiple branches in parallel, the last one pushed will squash even the build of other branches. Previously we only pushed images from `dev` branch which are not a pull request.

**Review by braktar — CHANGES_REQUESTED** [review]

File: .github/actions/build.sh (hunk 1 above)
Comment (alain-andre) [reply .github/actions/build.sh:9]: The last branch pushed overwrites the image generated by another branch, the mechanism is unchanged, do you want to change it ? In what way ?

The value of the TAG is the name of the owner as it has been defined for Travis. *RESPOSITORY* has a value of *Mapotempo/optimizer-ortools* for example in this project and so the TAG will have *Mapotempo* as its value.

Comment (alain-andre) [reply .github/actions/build.sh:9]: Ok. I changed the conditions. It will build and test it every time but will deliver only on dev branch or tag.

Comment (fonsecadeline) [reply .github/actions/build.sh:9]: If I am not mistaken, on Travis we used to be able to generate an image for any branch, not only _dev_ . This allowed us to run optimzer_api tests with the correct image without waiting our edits on optimizer-ortools side are merged.

This means we should deliver image not only from dev branch.

Comment (braktar) [reply .github/actions/build.sh:9]: We were only able to push on the repository from [dev branch](https://github.com/Mapotempo/optimizer-ortools/blob/22eeea03ddc1b60354cbae366fda44ab78732da3/.travis.yml#L12) or from a [tag](https://github.com/Mapotempo/optimizer-ortools/blob/22eeea03ddc1b60354cbae366fda44ab78732da3/.travis.yml#L37).
What we were able to do is point the images from our personal repositories or the one from mapotempo

**Review by braktar — APPROVED** [review]

**Review by senhalil — APPROVED** [review]

**Review by fonsecadeline — APPROVED** [review]

**Pull request merged**